src/components/entity/state-badge.html

```diff
@@ -61,79 +61,41 @@
     return window.hassUtil.computeDomain(stateObj);
   }
 
-  _rgbFromBrightness(brightness) {
-    // Set icon bbroghtness to a scale between 15% bright to 100% bright.
-    const iconBrightness = (brightness + 45) / (255 + 45);
-    const hostBrightness = 1 - iconBrightness;
-    const re = /rgb\((\d+), (\d+), (\d+)\)/;
-    const rgbIconMatch = window.getComputedStyle(this.$.icon).color.match(re);
-    const rgbHostMatch = window.getComputedStyle(this).color.match(re);
-    let rgb;
-    if (rgbIconMatch && rgbHostMatch) {
-      const rgbIcon = rgbIconMatch.slice(1).map(v => Number.parseInt(v, 10));
-      const rgbHost = rgbHostMatch.slice(1).map(v => Number.parseInt(v, 10));
-      rgb = [
-        Math.round((rgbIcon[0] * iconBrightness) + (rgbHost[0] * hostBrightness)),
-        Math.round((rgbIcon[1] * iconBrightness) + (rgbHost[1] * hostBrightness)),
-        Math.round((rgbIcon[2] * iconBrightness) + (rgbHost[2] * hostBrightness)),
-      ];
-      if (rgb.some(v => Number.isNaN(v))) {
-        return undefined;
-      }
-    }
-    return rgb;
-  }
 
-  /**
-   * Called when an attribute changes that influences the color of the icon.
-   */
-  updateIconColor(newVal) {
-    const icon = this.$.icon;
-    // hide icon if we have entity picture
-    if (newVal.attributes.entity_picture) {
-      this.style.backgroundImage = 'url(' + newVal.attributes.entity_picture + ')';
-      icon.style.display = 'none';
-      return;
-    }
+  updateIconAppearance(newVal) {
+    const iconStyle = {
+      display: 'inline',
+      color: '',
+      filter: '',
+    };
+    const hostStyle = {
+      backgroundImage: '',
+    };
+    /* eslint-disable no-labels, no-restricted-syntax */
+    calcProps: {
```

Comment:
I am not a fan of using labels. It makes for code that is very confusing to follow. What about

```js
if (newVal.attributes.entity_picture) {
  …
} else if (newValu.state !== 'unavailable') {
  …
}
```

Reply:
Then, in this case I'd end up with:
```js
if (newVal.attributes.entity_picture) {
  …
} else if (newVal.state !== 'unavailable') {
  …
} else {
  if (newVal.attributes.hs_color) {
    ..
  }
  if (newVal.attributes.brightness) {
    ..
  }
}
```
If that's fine I'll do it like that, I don't really care. But it's not more readable then the label imo.
What I like about the label/block approach is we can do some kind of guard clause in the block and we don't have to return from the function, and we don't have to define a seperate function for that block.

I could of course achieve the same with a self-invoking function, but is this more readable:
```js
const calcProps = function(){...}; calcProps()
```

Reply:
I don't mind the 2 level nested if statements. Instead of an inline defined function, what about you extract it as a function.

```js
const { iconStyle, hostStyle } = this._calcStlye(newVal);
```

And in `_calcStyle` you just return an object with those 2 keys.